experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -1042,20 +1034,24 @@ function extractContentMap<T extends InsertableTypedNode<MapNodeSchema>>(
 	};
 }
 
-function extractContentObject<T extends object>(input: T): ExtractedFactoryContent<T> {
-	const output: Record<string, unknown> = {};
+function extractContentObject(input: {
+	readonly [P in string]?: FactoryContent;
+}): ExtractedFactoryContent {
+	const output: Record<string, FactoryContent> = {};
 	if (typeNameSymbol in input) {
 		Object.defineProperty(output, typeNameSymbol, { value: input[typeNameSymbol] });
 	}
 	const hydrators: [key: string, hydrate: ProxyHydrator][] = [];
 	for (const [key, value] of Object.entries(input)) {
-		const { content: childContent, hydrateProxies } = extractFactoryContent(value);
-		output[key] = childContent;
-		hydrators.push([key, hydrateProxies]);
+		if (value !== undefined) {
```

Comment:
With the new typing, the type system objected to undefined here, so I added this guard to skip explicit undefined members. This impacted a test, but the change seems fine. The code receiving the extracted content object should be fine if optional fields are missing instead of explicitly undefined.

Reply:
I'll include this in my followup with the rename.